WIP: 2.1.1 puppeteer parity

Granitosaurus opened the pull request:
This is huge rework for catching up with puppeteer 2.1.1

For more see:
https://github.com/pyppeteer/pyppeteer2/projects/3

Mattwmaster58 commented on setup.py:
```diff
@@ -46,7 +46,8 @@
     'urllib3<1.25',
-    'tqdm'
+    'tqdm',
+    'aenum'
```
IMO, we should conditionally add `aenum` for Python versions < 3.4

Granitosaurus replied:
Actually `aenum` is required for every python version as the new Enum implementation doesn't support nested enums. We're trying to keep things as close to original puppetteer's code as possible. See related: https://bugs.python.org/issue33976

Mattwmaster58 replied:
Ahh, I see

Mattwmaster58 commented on setup.py:
```diff
@@ -46,7 +46,8 @@
     'urllib3<1.25',
-    'tqdm'
+    'tqdm',
+    'aenum'
```
Should we add this conditionally for Python versions < 3.4?

Mattwmaster58 commented:
<s>Blocker: undefined `computeQuadArea` in jshandle (see https://stackoverflow.com/questions/24467972 for easy implementation)</s>

Mattwmaster58 commented on pyppeteer/domworld.py:
```diff
@@ -0,0 +1,567 @@
+        context = await self.executionContext
+        document = await context.evaluateHandle('document')
+        self._documentFuture.set_result(document.asElement())
+        return await self._documentFuture
```
This would be more analogous to the JS source I'm pretty sure. This will allow for `_document` to be called twice but only return the same future like (i think) the js authors intended. No references to `_document` will have to be changed as it returns an Awaitable.
```suggestion
    @property
    def _document(self) -> Awaitable['ElementHandle']:
        if self._documentFuture:
            return await self._documentFuture

        async def _create_doc_future():
            context = await self.executionContext
            document = await context.evaluateHandle('document')
            return document.asElement()
        self._documentFuture = self._loop.create_task(_create_doc_future())
        return await self._documentFuture
```

Mattwmaster58 committed:
fix: better generated protocol formatting

Mattwmaster58 committed:
import sorting

Mattwmaster58 committed:
fix: erroneously imported statements

Mattwmaster58 committed:
test: add test_click.py

Mattwmaster58 committed:
fix: change broken content to set page too

Mattwmaster58 committed:
:bug: wip fix: failing accessibility tesst

Mattwmaster58 committed:
fix: typos in accessibility.py + failing accessibility tests

Mattwmaster58 committed:
fix: :white_check_mark: all tests passing for accessibility module

Mattwmaster58 committed:
:white_check_mark: fix: failing touchscreen test

Mattwmaster58 committed:
add test_keyboard.py tests, :white_check_mark: all passing

Commit:
add ported tests from puppeteer for page.click

Granitosaurus committed:
Update tests/test_click.py

Co-Authored-By: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

Commit:
refactored as per PR recommendations: remove '() => ' eval prefixes and server path '/' prefixes

Granitosaurus committed:
Merge pull request #78 from pyppeteer/test_click

add ported tests from puppeteer for page.click

Commit:
add tests for jshandle

Commit:
add frame tests and fix dumpFrames test utility function

Commit:
add frame tests and fix dumpFrames test utility function

Mattwmaster58 committed:
chore: cleanup up test implemnetations to better utilize pytest.raises

and other things:
 - remove useless req handlers

Mattwmaster58 committed:
Merge remote-tracking branch 'origin/pup2.1.1' into pup2.1.1

Mattwmaster58 committed:
fix: modify legacy tests so that they can at least be run

even though they fail, useful for when you want to run the whole suite

Mattwmaster58 committed:
fix: reduce timeout by a factor of 12x s

The slowest test runs in 5s, by setting the timeout to double, we shouldn't run into any issues with it and should speed releasing of test workers when a test is hanging

Mattwmaster58 committed:
ft: cleanup var setter implmentation

Mattwmaster58 committed:
ft: update addopts -n arg

still disabled ATM pending discussion

Mattwmaster58 committed:
fix: longstanding minor todos

Granitosaurus committed:
Update tests/test_jshandle.py

Co-Authored-By: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

VicGrygorchyk commented on pyppeteer/network_manager.py:
```diff
@@ -472,49 +425,35 @@ async def respond(self, response: Dict) -> None:  # noqa: C901
-        if response.get('body') and isinstance(response['body'], str):
-            responseBody: Optional[bytes] = response['body'].encode('utf-8')
+        if isinstance(response.get('body'), str):
+            responseBody: bytes = response['body'].encode('utf-9')
```
Is it intended to have encoding `utf-9`? Should we stick with `utf-8`?

Mattwmaster58 replied:
Your correct, definitely a typo, good catch. I wasn't aware that UTF-9 even existed but apparently it does:

> UTF-9 and UTF-18, despite being functional encodings, were April Fools' Day RFC joke specifications.

Mattwmaster58 committed:
fix: utf-9 -> utf-8 encoding

Mattwmaster58 committed:
chore: add todo item for using set to construct graph from set directly

Mattwmaster58 committed:
fix: correct assets location

Mattwmaster58 committed:
fix: correct typing of RemoteObject

Mattwmaster58 committed:
fix: analgous JS behaviour to replace bad tf-8 bytes

node's Buffer.toString will replace bad utf bytes with �, but python's will not unless you tell it too

VicGrygorchyk committed:
Fix typo.

VicGrygorchyk committed:
Copypaste js test sctructure.

Mattwmaster58 committed:
ft: migrate to using protocol class in type hint

VicGrygorchyk committed:
Merge branch 'pup2.1.1' of https://github.com/pyppeteer/pyppeteer2 into pup2.1.1

VicGrygorchyk committed:
Added test_click_document.

VicGrygorchyk committed:
Added test test_resize_textarea.

VicGrygorchyk committed:
Added test_select_text_with_mouse.

VicGrygorchyk committed:
Added test_trigger_hover_state.

VicGrygorchyk committed:
Added test_trigger_hover_removed_window_node.

Mattwmaster58 committed:
fix: implement tests for coverage + fix some problem w/ them

Mattwmaster58 committed:
fix: better reasoning for disposing connections

VicGrygorchyk committed:
Added test_tween_mouse_movement.

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_coverage

VicGrygorchyk committed:
Added test_mobile_viewport.

VicGrygorchyk committed:
Refactor imports.

Mattwmaster58 committed:
fix: better reasoning for disposing connections

Mattwmaster58 committed:
fix: occured -> occurred

VicGrygorchyk committed:
Fix test_set_modifier_keys_onclick.

VicGrygorchyk committed:
Del unused import.

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_coverage

Mattwmaster58 committed:
fix: correctly assign anonymous functions a debugging URL

Commit:
add some missing tests

Granitosaurus committed:
Merge pull request #79 from pyppeteer/test_jshandle

add tests for jshandle

Commit:
add OrderedSet

Commit:
fix elementhandle reference not being a property

Commit:
fix waitFor() method being incorrectly marked as async method

Commit:
fix wait task signature not including required loop argument

Commit:
ensure is_js_function is dealing with strings

Commit:
restructure waitForFunction signature to allow function argument passover

Commit:
restructure waitFor to first check wether string is a function before checking for selectors

VicGrygorchyk committed:
Use better test naming.

VicGrygorchyk committed:
Don't use anonymous func for oneliners.

Mattwmaster58 committed:
Merge pull request #85 from VicGrygorchyk/pup2.1.1

test_mouse.py implemented from mouse.spec.js

Mattwmaster58 committed:
Add a pre-commit config (#82)

For better code quality

Mattwmaster58 committed:
don't run check-builtin-literals twice

Mattwmaster58 committed:
fix: run pre-commit hooks for the first time

Mattwmaster58 committed:
fix: revamp tests for tracing

Mattwmaster58 committed:
chore: remove unused, internal tracing method

Mattwmaster58 committed:
fix: (actually) run precommit hooks for the first time

Somehow these slipped out, not sure how :|

Mattwmaster58 committed:
update pre-commit to account for cases where package isn't installed in pre-commit env

Mattwmaster58 committed:
add test stubs for requestinterception

Mattwmaster58 committed:
fix: correct setting of _interceptionId and _resourceType, improve _onRequestWillBeSent handling

Mattwmaster58 committed:
ft: implement methods needed for request redirection

Mattwmaster58 committed:
ft: implement some test methods

VicGrygorchyk committed:
Initial skeleton for tests.

VicGrygorchyk committed:
Add test_page_Jeval_method.

VicGrygorchyk committed:
Add test_Jeval_accepts_args.

VicGrygorchyk committed:
Add test_Jeval_accepts_elementHandle.

VicGrygorchyk committed:
Add test_Jeval_throws_exception.

VicGrygorchyk committed:
Add test_page_JJeval_method.

VicGrygorchyk committed:
Expand test test_page_JJeval_method.

VicGrygorchyk committed:
Add test_page_J_queries_element.

VicGrygorchyk committed:
Add test_page_JJ_queries_elements.

VicGrygorchyk committed:
Make pythonistic test_page_JJ_queries_elements.

VicGrygorchyk committed:
Add test_page_JJ_returns_empty_array.

VicGrygorchyk committed:
Add test_Jx_queries_elements and test_Jx_returns_empty_array.

VicGrygorchyk committed:
Add test_Jx_returns_multiple_elements.

VicGrygorchyk committed:
Add test_ElementHandle_J_queries_element.

VicGrygorchyk committed:
Add test_ElementHandle_J_returns_none_if_no_element.

VicGrygorchyk committed:
Add test_ElementHandle_Jeval_evalates_js_func.

VicGrygorchyk committed:
Add test_ElementHandle_Jeval_retrieves_content.

VicGrygorchyk committed:
Add test_ElementHandle_Jeval_raises_exc_if_no_selector.

VicGrygorchyk committed:
Add test_ElementHandle_JJeval_executes_js_func.

VicGrygorchyk committed:
Add test_ElementHandle_JJeval_retrieves_content.

VicGrygorchyk committed:
Add test_ElementHandle_JJeval_returns_empty_array_if_no_selector.

VicGrygorchyk committed:
Add test_ElementHandle_JJ_queries_elements.

VicGrygorchyk committed:
Shorten test names.

VicGrygorchyk committed:
Add test_ElementHandle_JJ_returns_empty_list_if_no_element.

VicGrygorchyk committed:
Add test_ElementHandle_Jx_queries_element.

VicGrygorchyk committed:
Add test_ElementHandle_Jx_returns_none_if_no_element.

VicGrygorchyk committed:
Fix naming.

Mattwmaster58 committed:
ft: work on cleaner server implementation

VicGrygorchyk committed:
Update table with reminders about methods.

Co-Authored-By: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

VicGrygorchyk committed:
Merge branch 'pup2.1.1' of https://github.com/pyppeteer/pyppeteer2 into pup2.1.1

VicGrygorchyk committed:
Merge branch 'pup2.1.1' of https://github.com/VicGrygorchyk/pyppeteer2 into pup2.1.1

VicGrygorchyk committed:
Use test classes.

VicGrygorchyk committed:
Use camelCase.

VicGrygorchyk committed:
Use match with pytest raise exc.

Mattwmaster58 committed:
ft: finish cleaning up adhoc server request handling

Mattwmaster58 committed:
ft: finish cleaning up adhoc server request handling

Mattwmaster58 committed:
fix: regression due to server app refactoring + updating of pytest raises assertions

Mattwmaster58 committed:
remove all # noqa: E501 comments

most have been rendered obsolete

Mattwmaster58 committed:
remove all # noqa: E501 comments

most have been rendered obsolete

Mattwmaster58 committed:
ft: implement some more tests

Mattwmaster58 committed:
fix: Response.buffer properly encoding response

Mattwmaster58 committed:
fix: remove erroneous statement

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_requestinterception

# Conflicts:
#	pyppeteer/network_manager.py

Mattwmaster58 committed:
ft: better explain why we hook --verbose pytest flag

Mattwmaster58 committed:
fix: double sending of Fetch.continueRequest when request interception is enabled

Mattwmaster58 committed:
fix: better adhoc server responses

Mattwmaster58 committed:
ft: add all tests for page.SetRequestInterception

Mattwmaster58 committed:
ft: add stubs for golden-* comparisons

Mattwmaster58 committed:
ft: refactor Request.abort/respond/continue to get rid of JS style argument

Mattwmaster58 committed:
ft: implement Request.respond and Request.continue_ tests

Mattwmaster58 committed:
fix: cleanup adhoc server methods

Mattwmaster58 committed:
fix: various typos in tests

Mattwmaster58 committed:
fix: properly implement custom, ad-hoc responses

Mattwmaster58 committed:
fix: more test typo corrections

Mattwmaster58 committed:
Merge branch 'pup2.1.1_test_tracing' into pup2.1.1

# Conflicts:
#	tests/test_tracing.py

Mattwmaster58 committed:
fix: more forgiving releaseObject

Mattwmaster58 committed:
fix: unskip test with now existing server side implmentation

VicGrygorchyk committed:
Refactor tests for aliases.

VicGrygorchyk committed:
Explain assert for aliases.

VicGrygorchyk committed:
port queryselector.spec.js to test_queryselectors.py (#94)

* Initial skeleton for tests.

* Add test_page_Jeval_method.

* Add test_Jeval_accepts_args.

* Add test_Jeval_accepts_elementHandle.

* Add test_Jeval_throws_exception.

* Add test_page_JJeval_method.

* Expand test test_page_JJeval_method.

* Add test_page_J_queries_element.

* Add test_page_JJ_queries_elements.

* Make pythonistic test_page_JJ_queries_elements.

* Add test_page_JJ_returns_empty_array.

* Add test_Jx_queries_elements and test_Jx_returns_empty_array.

* Add test_Jx_returns_multiple_elements.

* Add test_ElementHandle_J_queries_element.

* Add test_ElementHandle_J_returns_none_if_no_element.

* Add test_ElementHandle_Jeval_evalates_js_func.

* Add test_ElementHandle_Jeval_retrieves_content.

* Add test_ElementHandle_Jeval_raises_exc_if_no_selector.

* Add test_ElementHandle_JJeval_executes_js_func.

* Add test_ElementHandle_JJeval_retrieves_content.

* Add test_ElementHandle_JJeval_returns_empty_array_if_no_selector.

* Add test_ElementHandle_JJ_queries_elements.

* Shorten test names.

* Add test_ElementHandle_JJ_returns_empty_list_if_no_element.

* Add test_ElementHandle_Jx_queries_element.

* Add test_ElementHandle_Jx_returns_none_if_no_element.

* Fix naming.

* Update table with reminders about methods.

Co-Authored-By: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

* Use test classes.

* Use camelCase.

* Use match with pytest raise exc.

* Refactor tests for aliases.

* Explain assert for aliases.

Co-authored-by: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

Mattwmaster58 committed:
ft: HTTPS server

Mattwmaster58 committed:
ft: better url redirects

Commit:
Fix: Make task queue return results (#97)

Mattwmaster58 committed:
ft: re-implement test server

Mattwmaster58 committed:
ft: re-implement test server

Mattwmaster58 committed:
fix: send referrer NOT referer

Mattwmaster58 committed:
fix: asyncio.gather all subscribing functions to request path

Mattwmaster58 committed:
fix: add missing implementation of test now possilbe

Mattwmaster58 committed:
fix: add missing test implementations, delete test_network.py (superseded by test_navigation.py)

Mattwmaster58 committed:
fix: redirect chaining failing to create new scope

which resulted in every item of the redirect chain simply redirecting to the last one directly

Mattwmaster58 committed:
fix: failing+missimplemented tests

Mattwmaster58 committed:
fix: failing tests/implementations

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_requestinterception

Mattwmaster58 committed:
ft: add golden assets

Mattwmaster58 committed:
add utils for comparing images, text

Mattwmaster58 committed:
use correct kwarg for comparing imgs

Mattwmaster58 committed:
update test to await images byts

Mattwmaster58 committed:
add fixture for comparing images, text`

analogous to puppeteer's golden utils

Mattwmaster58 committed:
ft: complete isGolden fixture (write diff, actual, and expected)

Mattwmaster58 committed:
add pillow dep, update deps

Mattwmaster58 committed:
Merge branch 'dev' into pup2.1.1

# Conflicts:
#	pyppeteer/page.py

Mattwmaster58 committed:
Merge branch 'dev' into pup2.1.1

# Conflicts:
#	CONTRIBUTING.md
#	tox.ini

Mattwmaster58 committed:
complete migration to pyppeteer name

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_requestinterception

Mattwmaster58 committed:
fix bugs that occur upon a failing mismatch

Mattwmaster58 committed:
fix bugs that occur upon a failing mismatch

Mattwmaster58 committed:
revamp test_element_handle.py for pytesting

Mattwmaster58 committed:
fix bugs related to handle getting

Commit:
resolve conflicts

Granitosaurus committed:
Merge pull request #81 from pyppeteer/test_frames

Add tests for frames

Commit:
fix signature for page.waitForFunction

Commit:
Merge branch 'pup2.1.1' of github.com:pyppeteer/pyppeteer2 into pup2.1.1

Mattwmaster58 committed:
fix failure to not get a clickable point

Mattwmaster58 committed:
fix: change misleading error msg (incidental)

also: no clue where I fixed the bug where the bounding box was incorrect - I have a theory that it's a test server issue but have no proof as I can't get the test too fail and at this point I don't think it matters if it never fails again, but just in case I've added a note.

Mattwmaster58 committed:
fix: change misleading error msg (incidental)

Mattwmaster58 committed:
test: add disclaimer that test will fail with unordered set

Mattwmaster58 committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_element_handle

Mattwmaster58 committed:
ft: change to ordered_set (pure python) from orderedset (Cpython)

Mattwmaster58 committed:
fix: remove unecessary disclaimer

Zebradil committed:
Improve progress bar for chromium download

This shows progress in bytes, scaled as needed and disable progress bar
properly.

VicGrygorchyk committed:
Add emulation tests.

VicGrygorchyk committed:
Added test_supports_landscape_emulation.

VicGrygorchyk committed:
Fix test_page_emulates_mobile_devices.

VicGrygorchyk committed:
Test emulation media.

VicGrygorchyk committed:
Add TestEmulateMediaFeatures test_throws_err_if_bad_arg.

VicGrygorchyk committed:
Add test_emulate_timezone_works.

VicGrygorchyk committed:
Add test_err_thrown_if_invalid_timezone_ids.

VicGrygorchyk committed:
Fix test_is_detecteable_by_modernizr.

VicGrygorchyk committed:
Merge with original source.

VicGrygorchyk committed:
Make emulate method original puppeteer alike.

Zebradil committed:
Move tqdm import back to it's place

Mattwmaster58 committed:
Improve progress bar for chromium download (#113)

* Improve progress bar for chromium download

This shows progress in bytes, scaled as needed and disable progress bar
properly.

VicGrygorchyk committed:
Revert pyproject to pup2.1.1 version.

Mattwmaster58 committed:
use asyncio.gather for faster sending of emulation command (#117)

VicGrygorchyk committed:
Merge branch 'pup2.1.1' of https://github.com/pyppeteer/pyppeteer into pup2.1.1

Commit:
fix typo in backendNodeId argument

Granitosaurus committed:
Merge pull request #110 from pyppeteer/pup2.1.1_test_element_handle

revamp test_element_handle

Granitosaurus committed:
Merge branch 'pup2.1.1' into pup2.1.1_test_coverage

Granitosaurus committed:
Merge pull request #86 from pyppeteer/pup2.1.1_test_coverage

revamp test_coverage.py

Mattwmaster58 committed:
ft: merge #110

Commit:
fix execution context adoptElementHandle method not unpacking received object from the browser

VicGrygorchyk committed:
Consistency with quotes.

VicGrygorchyk committed:
Use is for bool comparision.

VicGrygorchyk committed:
Fix comment to TestEmulationMedia.

VicGrygorchyk committed:
Copy test queryselector from source branch.

VicGrygorchyk committed:
Copy poerty.lock and pyproject.toml from source branch.

VicGrygorchyk committed:
Remove extra blank line.

VicGrygorchyk committed:
Remove extra goto.

VicGrygorchyk committed:
Implement page.emulateMediaType().

VicGrygorchyk committed:
Use warning instead of raise exception.

VicGrygorchyk committed:
Fix docstring

Co-authored-by: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

Mattwmaster58 committed:
implement Page.emaulteMediaType (#120)

Implement Page::emulateMediaType() from @VicGrygorchyk

VicGrygorchyk committed:
Merge branch 'pup2.1.1' of https://github.com/pyppeteer/pyppeteer into pup2.1.1

VicGrygorchyk committed:
Test deprecated emulateMedia() as a part of TestEmulateMediaType suite.

VicGrygorchyk committed:
Dont compare with True.

VicGrygorchyk committed:
Merge emulataion media tests into one.

Mattwmaster58 committed:
Add emulation tests (#116) 

Add tests for emulation

MorbidAngel committed:
fix: wait for Target.fetchFromTarget command

MorbidAngel committed:
Merge remote-tracking branch 'origin/pup2.1.1' into fix-wait-for-detach-from-target-command

Mattwmaster58 committed:
fix: use absolute value for area

Mattwmaster58 committed:
fix: fix arrow function syntax error

MorbidAngel committed:
Update pyppeteer/page.py

Co-authored-by: Mattwmaster58 <26337069+Mattwmaster58@users.noreply.github.com>

Mattwmaster58 committed:
async onTargetDetached (#130)

fix: wait for Target.fetchFromTarget command

Mattwmaster58 committed:
Merge remote-tracking branch 'origin/pup2.1.1' into pup2.1.1

jessrosenfield committed:
Fix asyc call to waitTask.rerun

Mattwmaster58 committed:
Merge pull request #140 from jessrosenfield/pup2.1.1

Fix asyc call to waitTask.rerun

Mattwmaster58 committed:
fix: broken _go implementation

ps: I actually don't really know why this worked when other stuff didn't, the reason I changed it was on a hunch that making it more like puppeteer's implementation would fix things (and it did)

Mattwmaster58 committed:
ft: add custom headers stub infra

Mattwmaster58 committed:
ft: one_time_request_delay method

Mattwmaster58 committed:
ft: more ad-hoc custom header infra

Mattwmaster58 committed:
fix: faulty tests causing failures

Mattwmaster58 committed:
fix: test_prioritizes_nav_timeout_of_default

Mattwmaster58 committed:
fix: navigate to the correct path for some tests

Mattwmaster58 committed:
fix: look for substr in page url, not response url

Mattwmaster58 committed:
fix: correct frame rejection logic

Mattwmaster58 committed:
fix: allow setting of content type, default to text/html

otherwise, the browser will fail to a page returning a non-html response

Mattwmaster58 committed:
fix: waitForNavigation fails on detached frame implementation fix

the last implementation didn't even make sense

Mattwmaster58 committed:
fix: revert unconditional skip

I'm not sure why I skipped this in the first place but my best guess is that the test server implementation was faulty causing this test to fail and I thought it was the test's fault. Now that the test server is A-OK, this test is unskipped

Mattwmaster58 committed:
fix: specify waittarget so that testing of domcontentloaded and load

Mattwmaster58 committed:
fix: don't re-use reponses from other requests

this is why that one test before was skipped: the test prior used used a 204, but then the next test also used a 204, but you can't use the same response twice. this changes things so each response is generated as needed

Mattwmaster58 committed:
ft: don't use lambda to create new class instances, just refer to the class itself

Mattwmaster58 committed:
hotfix: gross fix for multiprocessing

polyfloyd committed:
Expose the securityState of the Response

Commit:
update browser fetcher to handle PYPPETEER_CHROMIUM_REVISION

Commit:
fix autoClose not working

polyfloyd committed:
Restore only the signal handlers that were altered (#159)

polyfloyd committed:
Expose the sanList property of SecurityDetails

VicGrygorchyk committed:
Pass params for set_one_time_response.

VicGrygorchyk committed:
Add description.

Mattwmaster58 committed:
Merge remote-tracking branch 'origin/pup2.1.1' into pup2.1.1

Sunda001 commented:
hmm

Mattwmaster58 committed:
Merge pull request #164 from polyfloyd/security-fields

Expose more security related fields

Mattwmaster58 committed:
Merge pull request #170 from VicGrygorchyk/add_kwargs_for_onetime_test_server_response

Add kwargs for onetime test server response

Mattwmaster58 committed:
merge branch dev

polyfloyd committed:
Do not try to set an exception on finished futures

This fixes any InvalidState exceptions that occurred when terminating
with an unrelated error.

icemonster commented on pyppeteer/execution_context.py:
```diff
@@ -122,117 +109,48 @@ def _convertArgument(self, arg: Any) -> Dict:  # noqa: C901
+
+
+def rewriteError(error: Exception) -> Union[None, Dict[str, Dict[str, str]]]:
+    msg = error.args[0]
```
Issue #119 
We probably want to check if `error.args` has any elements, otherwise this throws an exception

Mattwmaster58 replied:
I agree! We're accepting PRs against this branch if you'd like to make the necessary changes

Mattwmaster58 committed:
Merge pull request #216 from polyfloyd/fix-invalidstate-exceptions

Do not try to set an exception on finished futures

scratchmex commented:
Legit question: why is 2.1.1 the target version if puppeteer now is in version 10?

Mattwmaster58 commented:
When the project started, 2.1.1 was the current version. We're behind

On Sat., Sep. 11, 2021, 10:07 a.m. Ivan Gonzalez, ***@***.***>
wrote:

> Legit question: why is 2.1.1 the target version if puppeteer now is in
> version 10?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyppeteer/pyppeteer/pull/16#issuecomment-917431249>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGI56LIZPVDDE42YFA45GETUBN5DZANCNFSM4KZE5GZA>
> .
>

miracle2k commented:
Could the page make it a little bit more clear what the status of this branch is vs "dev", and which one is recommended that people a) user as end users and b) send patches to? It seems `dev` is very active, while this port is not.

Mattwmaster58 committed:
Merge remote-tracking branch 'origin/pup2.1.1' into pup2.1.1
